src/Symfony/Component/Messenger/Command/FailedMessagesRetryCommand.php

```diff
@@ -39,14 +40,16 @@ class FailedMessagesRetryCommand extends AbstractFailedMessagesCommand
     private $eventDispatcher;
     private $messageBus;
     private $logger;
+    private $globalReceiverName;
 
-    public function __construct(string $receiverName, ReceiverInterface $receiver, MessageBusInterface $messageBus, EventDispatcherInterface $eventDispatcher, LoggerInterface $logger = null)
+    public function __construct(?string $globalReceiverName, ?ReceiverInterface $globalReceiver, MessageBusInterface $messageBus, EventDispatcherInterface $eventDispatcher, LoggerInterface $logger = null, ?ServiceLocator $failureTransports = null)
```

Comment:
Could we eliminate the new `$failureTransports` argument and instead remove the type-hint from `?ReceiverInterface $globalReceiver`? The 2nd arg would then be able to receive a `ReceiverInterface` OR a `ServiceLocator` and we would check the type and act accordingly. Maybe we would just pass that straight to the parent constructor, since it already has the deprecation.